### DISCUSSION TITLE
Focusrite driver issue

### DISCUSSION THREAD

**Discussion by jgold55** [jamulussoftware/jamulus#1484] [Technical Support & Help]
Total newbie here, so forgive me if I'm in the wrong forum.  
I ran Jamulus for the first time and it screwed up my music mixing software (Sonar, Focusrite 18i20).  I get visible sound levels but no audio output.  What did it do to my driver? How can I fix it?

**Comment by ann0see** [comment]
Hi @jgold55 !

Sorry to hear that!

> Total newbie here, so forgive me if I'm in the wrong forum.

No problem. Bug investigation is ok but I moved it to Technical Support and Help since I think it's more of a support question.

>  I ran Jamulus for the first time and it screwed up my music mixing software (Sonar, Focusrite 18i20). I get visible sound levels but no audio output. 

Ok. So this means that you get audio into your software but not out of it. So what I would check first: What happens if you use the driver in Jamulus? If you can't hear sound there, I'd check if your headphones are working, turning up the volume (in the system and on your interface), ... If you can hear sound there, something might be wrong with your music mixing software: did you choose the correct inputs/outputs; higher buffer sizes, WDM/non ASIO APIs.,... 

> What did it do to my driver? 

It's highly unlikely that Jamulus or your music software changed anything in the driver. Both programs "control" the driver but don't usually damage it. Normally you can't damage software (worst would be a total reset of your driver or a new installation of it).

Another thought: 

Can you check if the device works on another device?

**Reply by jgold55** [reply]
Things seem ok with my mixing software now.  A reboot of everything helped.  
1. Jamulus worked fine (great even!), so the sound card/driver worked with that.
2.  When I finished with Jamulus and closed it, my mixing software would not work with the sound card.
It seems the sound card/driver will only work with one program at a time, and even when closed, Jamulus held on to the driver.
Does this make any sense? 
A related question would be - How can I run Jamulus and mixing software (which uses the same sound card) at the same time?
Joel

**Reply by ann0see** [reply]
> It seems the sound card/driver will only work with one program at a time, and even when closed, Jamulus held on to the driver.

As far as I know, ASIO only supports one program at a time. You could however try something like JACK router or VoiceMeeter to overcome this issue.

**Reply by jgold55** [reply]
Then, how can I play my guitar and sing at the same time? I have to use one mic for both?  
Or - My sound card has several inputs.  I guess if Jamulus uses the sound card output, I can set the relative input levels for voice and guitar using my ASIO control panel.
If I cant use my mixing/recording software because it wont share the driver, how can I record the Jamulus session?
 thanks,
Joel

**Reply by dcorson-ticino-com** [reply]
Jamulus has two channels, left and right.  You can use the possibilities of your sound card or the input pan control to set the relative volumes.
For recording you will need to use a digital cable to connect Jamulus to your software,
The only thing I have found that works on Win10 (without adding undue latency) appears to be Jack2 and JackRouter. For some reason JackRouter has been removed from the Jack2 package so you need to download it separately. (Note that I haven't checked for a couple of months now, maybe there are new versions which are simpler to use in the mean time)
See the Jack instructions for Win10 here: https://jackaudio.org/faq/jack_on_windows.html
These instructions assume that JackRouter is installed with Jack2, but in the latest version 1.9.16 it isn't. You need to download JackRouter separately from here: https://github.com/jackaudio/jack-router/tree/main/binaries
Once all is installed start the Jack server, start Jamulus and your application and tell them both to use JackRouter in their corresponding audio setups. Then you can connect them using the Jack audio connection kit. Oh yes, Jamulus must be connected to a server for JackRouter to allow connections.
There are lots of hoops you need to jump through to get it going, but in the end it worked for me.

**Reply by jgold55** [reply]
Interesting.. 
Or, I perhaps I can connect a handheld recorder to my other headphone output?

**Reply by ann0see** [reply]
The question is what you want to do. If you'd just like to record your performance, you could use the Jamulus recording feature: you'll need to set the `-R` argument on the server; see: https://jamulus.io/wiki/Command-Line-Options

**Reply by jgold55** [reply]
I'd like to record everybody, not just me.
It sounds like that is a command-line argument only, and is not available in the regular program?  
Does it save a wav file?  The instruction says it is a path name.  Where do you declare a file name? 
Danke

**Reply by ann0see** [reply]
You can also set it in the GUI. See: https://jamulus.io/de/wiki/Server-Win-Mac#recording

Hopefully it explains what you need. Yes, the files are in .wav format

**Comment by dcorson-ticino-com** [comment]
Hi jg , 
Did you connect to a server? You will not hear anything in Jamulus if you don't connect to a server. 

Note that Jamulus does not change anything in your driver by itself, but it can call your driver to let you change things there.

I would also start by trying Jamulus with your hardware (and it's ASIO driver if on Win10), without any other software.
Many people are successfully using hardware from Focusrite with Jamulus.
Once you know that works you can add other software components.
Let us know how that goes,
Don

**Reply by jgold55** [reply]
See my earlier reply.  Jamulus worked fine with the card.  After closing Jamulus my mixing software wouldnt work with the sound card.
